### PR TITLE
Handle new Cromwell logs output

### PR DESCRIPTION
**Pull request by jmthibault79**
DSDEEPB-2124

This will fix calls to
  /api/workspaces/{prj}/{wk}/submissions{subId}/workflows/{wfId}/outputs
which is currently failing because it's looking for the old format
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala [hunk 1]
```diff
@@ -468,9 +474,9 @@ class MockExecutionServiceDAO() extends ExecutionServiceDAO {
   }
 
   override def logs(id: String, userInfo: UserInfo) = Future.successful(ExecutionServiceLogs(id,
-    Map("x" -> Seq(Map(
-      "stdout" -> "gs://cromwell-dev/cromwell-executions/wf/this_workflow_exists/call-x/job.stdout.txt",
-      "stdout" -> "gs://cromwell-dev/cromwell-executions/wf/this_workflow_exists/call-x/job.stdout.txt")))))
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala:473]: This doesn't make sense, so I assume it was never used anywhere.  I updated it anyway.

File: src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala [hunk 2]
```diff
@@ -95,18 +95,24 @@ class SubmissionSpec(_system: ActorSystem) extends TestKit(_system) with FlatSpe
...
+              stderr = "gs://cromwell-dev/cromwell-executions/wf/this_workflow_exists/call-y/job.stderr-1.txt",
+              backendLogs = Some(Map(
+                "log" -> "gs://cromwell-dev/cromwell-executions/wf/this_workflow_exists/call-y/jes.log",
+                "stdout" -> "gs://cromwell-dev/cromwell-executions/wf/this_workflow_exists/call-y/jes-stdout.log",
+                "stderr" -> "gs://cromwell-dev/cromwell-executions/wf/this_workflow_exists/call-y/jes-stderr.log"
+              ))
+            ),
+            ExecutionServiceCallLogs(
+              stdout = "gs://cromwell-dev/cromwell-executions/wf/this_workflow_exists/call-y/job.stdout-2.txt",
+              stderr = "gs://cromwell-dev/cromwell-executions/wf/this_workflow_exists/call-y/job.stderr-2.txt")
```
Comment (MatthewBemis) [inline src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala:109]: Why doesn't this one also have backendLogs?

File: src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala (hunk 1 above)
Comment (MatthewBemis) [inline src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala:473]: Should it be removed then?

**Comment by MatthewBemis** [comment]
Other than the few very minor questions, :+1:

File: src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala (hunk 2 above)
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala:109]: To demonstrate that it is optional

**Comment by dvoet** [comment]
:+1:

File: src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala (hunk 1 above)
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSpec.scala:473]: Probably.  I think I'll do that in a different PR however.

**Pull request merged**